### DISCUSSION TITLE
Fixing Gaps When Filtering PartyFinder (PartyFinderGui)

### DISCUSSION THREAD

**Discussion by doggjoe9** [goatcorp/Dalamud#1107] [Ideas]
I am relatively new to the Dalamud internals so forgive me if I am misunderstanding anything. I noticed while developing a plugin that if you set the visibility of party finder listings, the remaining listings will still appear on their original page instead of being pushed to the front page.

I decided to see if I was just missing something; so I cloned ascclemens's repo ([BetterPartyFinder](https://git.anna.lgbt/ascclemens/BetterPartyFinder/src/branch/main)) to see if they had any better luck than me figuring it out. Lo and behold, after modifying it a bit to get it to actually start, I set the Min & Max ilvl requirements to 999 and still the remaining listings were spread out across multiple pages.

I checked the source for PartyFinderGui and looked at how it hides the entries which are set invisible during the ReceiveListing invokation.  [During this loop](https://github.com/goatcorp/Dalamud/blob/master/Dalamud/Game/Gui/PartyFinder/PartyFinderGui.cs/#L104)
```
for (var i = 0; i < packet.Listings.Length; i++)
{
    // these are empty slots that are not shown to the player
    if (packet.Listings[i].IsNull())
    {
        continue;
    }
    
    var listing = new PartyFinderListing(packet.Listings[i]);
    var args = new PartyFinderListingEventArgs(packet.BatchNumber);
    this.ReceiveListing?.Invoke(listing, args);
    
    if (args.Visible)
    {
        continue;
    }
    
    // hide the listing from the player by setting it to a null listing
    packet.Listings[i] = default;
    needToRewrite = true;
}
```
the hidden listings are set to a null listing to prevent them from being displayed.

I am wondering if `packet.Listings` were sorted so that the visible listings came first in the list and the invisible (null) listings came after, would this fix the issue and display them in a more regular way on the Party Finder window? This method certainly would avoid under/overwriting the allocated memory, since the listings are only permuted. My concern is that I cannot say I understand the whole breadth of this system in the larger context, so perhaps there might be another part of the Party Finder system that wants these listings in their original order.

Any input from experts would be appreciated. Apologies that I am not willing to rebuild Dalamud with this change to test it out, but that seems like a tall order to me, and I am also apprehensive because, like I said before, I am not well versed on what consequences come from reordering the packet listings.

Edit: Fixed the hyperlinks.

**Comment by Caraxi** [comment]
This cannot happen any other way due to the way the server sends the information to the client

**Reply by doggjoe9** [reply]
Oh I see. They are only sent one page at a time. Thanks for the quick response.
